src/Castle.Core.Tests/DynamicProxy.Tests/BasePEVerifyTestCase.cs

```diff
@@ -55,18 +52,13 @@ private static string FindPeVerifyPath()
 					return file;
 				}
 			}
-			throw new FileNotFoundException(
-				"Please check the PeVerifyProbingPaths configuration setting and set it to the folder where peverify.exe is located");
+			return null;
```

Comment:
This is the one part in this PR that I slightly worry about&mdash;it means that if you build Castle.Core on a system where PEVerify could be available, but isn't found, you won't even get a warning that tests will be less thorough. I tried replacing the `throw` with an `Assert.Warn` but that unfortunately didn't do anything.

Reply:
Your comment in another issue about providing the path to peverify.exe via an environment variable would be the best way to do this, that means we can fail the .NET Framework CI build inside build scripts if it can't find it, and then the unit test code doesn't have the massive list of Windows SDK paths. Then if we do add ILVerify we can do it clean too.

Happy for that to be done in another pull request, maybe even after we move to GitHub Actions where we can set up a matrix build.

Reply:
Glad to hear you're OK with finalizing this the proper way in a separate PR. To give some background, I didn't go straight there because I foresaw problems with IDE integration&mdash;right now, PEVerify is invoked even when you run tests from inside Visual Studio, and I wasn't sure whether this behaviour could be preserved... I expect this will take a little longer due to some MSBuild fiddling.

I'll definitely revisit this during the switch to GitHub Actions (which perhaps isn't very far off now).